### PR TITLE
Inspect: guard null plugin page hook in Async_Options::add_to_plugin_page()

### LINKED ISSUE #52424: Inspect: Async_Options::add_to_plugin_page() calls add_action() with a null hook on admin-ajax (PHP 8.5 deprecation flood)
### Impacted plugin

Inspect

### Quick start

Jetpack Inspect 2.0.0-alpha (trunk, build 35163429039), PHP 8.5 (also reproduces on 8.4 without the message), WordPress 7.2 nightly.

### What I expected

`Async_Options::setup()` on `admin_init` should only hook the options script when the plugin page hook exists.

### What happened

On every `admin-ajax.php` request (Action Scheduler runner, Wordfence scan, any plugin's ajax) PHP 8.5 logs:

```
PHP Deprecated: Using null as an array offset is deprecated, use an empty string instead in wp-includes/plugin.php on line 125
```

Stack:

```
wp-includes/plugin.php:125 add_filter()
wp-includes/plugin.php:451 add_action()
jetpack-inspect/packages/Async_Option/Async_Options.php:55 add_to_plugin_page()
jetpack-inspect/packages/Async_Option/Async_Options.php:76 setup()
jetpack-inspect/options.php:40 {closure} (admin_init)
```

`get_plugin_page_hook( 'jetpack-inspect', 'admin' )` returns `null` in the admin-ajax context (no menu is registered there), and `add_to_plugin_page()` passes that straight to `add_action()`, so core does `$wp_filter[ null ]`. On three sites this produced ~2,900 deprecation entries in 12 hours.

### Steps to reproduce

1. PHP 8.5, Jetpack Inspect active, `WP_DEBUG_LOG` on.
2. Load `wp-admin/admin-ajax.php?action=heartbeat` (or let any cron/ajax run).
3. See the deprecation above.

### Suggested fix

```php
public function add_to_plugin_page( $plugin_page, $parent_page ) {
    $plugin_page_hook = get_plugin_page_hook( $plugin_page, $parent_page );
    if ( ! $plugin_page_hook ) {
        return;
    }
    add_action( $plugin_page_hook, array( $this, '_print_options_script_tag' ) );
}
```

The same `Async_Option` package is vendored in other Jetpack plugins (Boost), so the fix belongs in the package.

### Site owner impact

Log noise only, but at every ajax request; PHP 9 will turn it into an error.

### PR DESCRIPTION
**Pull request by HFigarella**
Fixes #52424

## Proposed changes
* `Async_Options::add_to_plugin_page()` calls `get_plugin_page_hook()` on `admin_init` and passes the result straight to `add_action()`. On `admin-ajax.php` (Heartbeat, Action Scheduler runner, any admin AJAX) and other admin requests where the admin menu has not been built, `get_plugin_page_hook()` returns `null`, so `add_action( null, ... )` runs on every such request.
* On PHP 8.5 that null hook name triggers `Using null as an array offset` (deprecated) in `wp-includes/plugin.php` on every admin-ajax request. On PHP 8.4 the same call runs silently but still registers a callback under an empty hook.
* This PR returns early when there is no hook to attach to. No behaviour change on the actual plugin page, where the hook is always resolved.

## Related product discussion/links
* #52424

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions
* Run WordPress on PHP 8.5 with `WP_DEBUG_LOG` on and Jetpack Inspect active.
* Load any wp-admin page (Heartbeat fires `admin-ajax.php`), or run `wp eval 'do_action("admin_init");'` with `WP_ADMIN` defined and `DOING_AJAX` true.
* Before: the debug log fills with `PHP Deprecated: Using null as an array offset in wp-includes/plugin.php:125` (one per admin-ajax request).
* After: no deprecation; the Inspect settings page still receives its options `<script>` tag as before.

### CODE REVIEW

File: projects/plugins/inspect/changelog/fix-async-options-null-hook [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Significance: patch
+Type: fixed
+Comment: Async_Options: do not call add_action() with a null hook when get_plugin_page_hook() returns null (admin-ajax and other requests without an admin menu). Fixes a PHP 8.5 "Using null as an array offset" deprecation flood.
```
Comment (xavier-lc) [inline projects/plugins/inspect/changelog/fix-async-options-null-hook]: The `Comment:` field is free-form text for other developers and is not used by Changelogger (see `projects/packages/changelogger/README.md`), so as written this entry never reaches `CHANGELOG.md` / `readme.txt`. Since this is a fix a site owner would notice (debug-log flood on PHP 8.5), it needs an actual entry body: a blank line after the headers, then the text.

```suggestion
Significance: patch
Type: fixed

Stop registering an admin hook with a null name on admin-ajax requests, which flooded the debug log with a "Using null as an array offset" deprecation on PHP 8.5.
```

(`jp changelog add plugins/inspect -s patch -t fixed -e "…"` produces the right shape if you'd rather regenerate it.)

**Review by xavier-lc — COMMENTED** [review]
The fix itself looks right: `wp-admin/admin.php` loads `menu.php` before firing `admin_init`, while `admin-ajax.php` never builds the menu, so `get_plugin_page_hook()` returns `null` there and the guard is exactly what's needed. No behaviour change on the real settings page, where the hook is registered before `admin_init`.

One thing to fix before merge: the changelog entry is in the `Comment:` field, which Changelogger ignores, so nothing would land in the changelog. Inline suggestion below.

**Comment by xavier-lc** [comment]
Follow-up on test coverage: this is easy to lock in with a regression test, and I've verified the one below fails on trunk's `Async_Options.php` and passes with this PR's guard.

The catch is that `plugins/inspect` has a PHPUnit harness but no tests and no WordPress in its bootstrap, and `add_to_plugin_page()` needs `get_plugin_page_hook()` / `add_action()`. The monorepo answer is WorDBless via `automattic/jetpack-test-environment` (same as `starter-plugin`, `boost`, and ~49 other projects), so three small pieces:

**1. `composer.json`** — add to `require-dev` (then `jp install plugins/inspect` to refresh the lock):

```json
"automattic/jetpack-test-environment": "@dev"
```

**2. `tests/php/bootstrap.php`** — after the autoloader `require_once`:

```php
\Automattic\Jetpack\Test_Environment::init();
```

**3. `tests/php/packages/Async_Option/Async_Options_Test.php`**:

```php
<?php

namespace Automattic\Jetpack\Packages\Async_Option;

use PHPUnit\Framework\Attributes\CoversClass;
use WorDBless\BaseTestCase;

/**
 * @covers \Automattic\Jetpack\Packages\Async_Option\Async_Options
 */
#[CoversClass( Async_Options::class )]
class Async_Options_Test extends BaseTestCase {

	const PAGE_HOOK = 'admin_page_jetpack-inspect';

	public function set_up() {
		require_once ABSPATH . 'wp-admin/includes/plugin.php';
		remove_all_actions( self::PAGE_HOOK );
		unset( $GLOBALS['wp_filter'][''] );
	}

	private function make_instance() {
		return new Async_Options( 'jetpack-inspect-main', Registry::get_instance( 'jetpack_inspect' ) );
	}

	public function test_registers_callback_on_the_resolved_page_hook() {
		// A registered menu page is what makes get_plugin_page_hook() resolve.
		add_action( self::PAGE_HOOK, '__return_null' );
		$instance = $this->make_instance();

		$instance->add_to_plugin_page( 'jetpack-inspect', 'admin' );

		$this->assertSame( 10, has_action( self::PAGE_HOOK, array( $instance, '_print_options_script_tag' ) ) );
	}

	public function test_registers_nothing_when_the_page_hook_is_unresolved() {
		$instance = $this->make_instance();

		$instance->add_to_plugin_page( 'jetpack-inspect', 'admin' );

		$this->assertFalse( has_action( self::PAGE_HOOK, array( $instance, '_print_options_script_tag' ) ) );
		$this->assertArrayNotHasKey( '', $GLOBALS['wp_filter'], 'add_action() was called with an empty hook name' );
	}
}
```

Pre-registering an action on the page hook is what `get_plugin_page_hook()` actually checks (`has_action()`), so the test doesn't need to build a real menu or fake a capable user. Without the guard, the second test fails on the `''` key in `$wp_filter` on PHP ≤ 8.4, and on 8.5 it fails a second way too since the plugin's phpunit config sets `failOnDeprecation="true"`. PHPCS is clean from the monorepo root.

Standing up the test environment is a bigger diff than the fix itself, so I'm fine with either including it here or landing it as a follow-up — your call.

**Comment by HFigarella** [comment]
Thanks for the review, @xavier-lc. Both points are addressed:

- **Changelog:** the entry is now in the entry body so Changelogger picks it up (eef37a5).
- **Regression test:** added the WorDBless test as suggested, with the `automattic/jetpack-test-environment` dev dependency, the bootstrap init, and a refreshed `composer.lock` (9e53654). Locally on PHP 8.4 both tests pass, and the null-hook test fails when the guard is removed.

The Tests / Linting / Build / PHP Compatibility workflows are showing `action_required` because this is a first-time-contributor fork, so they haven't run on the latest push. Could a maintainer please approve the workflow run? Thanks!

File: projects/plugins/inspect/tests/php/packages/Async_Option/Async_Options_Test.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Automattic\Jetpack\Packages\Async_Option;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use WorDBless\BaseTestCase;
+
+/**
+ * @covers \Automattic\Jetpack\Packages\Async_Option\Async_Options
+ */
+#[CoversClass( Async_Options::class )]
```
Comment (xavier-lc) [inline projects/plugins/inspect/tests/php/packages/Async_Option/Async_Options_Test.php:11]: Optional: no test in the monorepo uses `@covers` or `#[CoversClass]` — coverage comes from the phpunit `<source>` include, so both are inert here. The `@covers` was in my snippet and PHPCBF turned it into the attribute pair; feel free to drop both to match the rest of the tree.

```suggestion
use WorDBless\BaseTestCase;

```

**Review by xavier-lc — APPROVED** [review]
Thanks, both points are addressed and this is good to go.

Re-verified at 9e53654: the test passes locally and fails when the guard is reverted to trunk, `composer install` from the committed lock produces no drift (the `jetpack-test-environment` reference matches `starter-plugin` and `boost`), and CI is green across PHP 7.4–8.5 — the 8.5 run is the direct proof the deprecation is gone, since the plugin's phpunit config has `failOnDeprecation="true"`.

One optional, non-blocking nit inline, which is on me since it came from the snippet I posted.
